src/test/typescript/typescript-parser_test.ts

```diff
@@ -0,0 +1,183 @@
+      assert.instanceOf(document, ParsedTypeScriptDocument);
+      assert.equal(document.url, '/typescript/test.ts');
+      console.log(document);
+      const program = document.program;
+      console.log(program.getRootFileNames());
+      const emitResult = program.emit();
+      const allDiagnostics =
+          ts.getPreEmitDiagnostics(program).concat(emitResult.diagnostics);
+
+      if (allDiagnostics) {
+        allDiagnostics.forEach((diagnostic) => {
+          // const { line, character } =
+          // diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start);
+          const message =
+              ts.flattenDiagnosticMessageText(diagnostic.messageText, '\n');
+          console.log(message);
+          // console.log(`${diagnostic.file.fileName}: ${message}`);
+        });
+      }
+
+      // assert.equal(document.ast.type, 'Program');
+      if (emitResult.emitSkipped) {
+        console.log('emitSkipped');
+      } else if (emitResult.emittedFiles) {
+        emitResult.emittedFiles.forEach((file) => {
+          console.log(file);
+        });
+      }
+    });
+
+    test.skip('basic compiler host', () => {
+      const testSource = `
+        class Foo extends HTMLElement {
+          bar: string = 'baz';
+        }
+      `;
+      const host: ts.CompilerHost = {
+        getSourceFile(
+            fileName: string,
+            languageVersion: ts.ScriptTarget,
+            onError?: (message: string) => void): ts.SourceFile /**/ {
+          console.log('getSourceFile', fileName); if (fileName === 'test.ts') {
+            return ts.createSourceFile(fileName, testSource, languageVersion);
+          } if (onError) onError('not found');
+          // tsc's declarations don't support strict null checks
+          return null as any as ts.SourceFile;
+        },
+        getDefaultLibFileName: () => '',
+        writeFile: (_fileName, _content) => {},
+        getCurrentDirectory: () => ts.sys.getCurrentDirectory(),
+        getCanonicalFileName: fileName => ts.sys.useCaseSensitiveFileNames ?
+            fileName :
+            fileName.toLowerCase(),
+        getNewLine: () => ts.sys.newLine,
+        useCaseSensitiveFileNames: () => ts.sys.useCaseSensitiveFileNames,
+        fileExists: (fileName: string) => fileName === 'test.ts',
+        readFile(fileName: string): string /**/ {
+          if (fileName === 'test.ts') {
+            return testSource;
+          } return null as any as string;
+        },
+        resolveModuleNames(_moduleNames: string[], _containingFile: string):
+            ts.ResolvedModule[] /**/ {/**/ throw new Error('unsupported');
+                                      /**/},
+        getDirectories(_path: string):
+            string[] /**/ {/**/ throw new Error('unsupported'); /**/}
+      };
+      const program = ts.createProgram(['test.ts'], {allowJs: true}, host);
+      console.log(program);
+    });
+
+    // test('throws syntax errors', () => {
```

Comment:
I hadn't ported these two yet. Since they're the same as the JS counterpoints, I just removed.